datalab/datalab.sh

```diff
@@ -28,53 +38,65 @@ function update_apt_get() {
   return 1
 }
 
-ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
-PROJECT=$(/usr/share/google/get_metadata_value ../project/project-id)
-DOCKER_IMAGE=$(/usr/share/google/get_metadata_value attributes/docker-image || true)
-SPARK_PACKAGES=$(/usr/share/google/get_metadata_value attributes/spark-packages || true)
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
+  return 1
+}
 
 if [[ "${ROLE}" == 'Master' ]]; then
-  update_apt_get
+  update_apt_get || err 'Failed to update apt-get'
+  mkdir -p "${DATALAB_DIR}"
+
   apt-get install -y -q docker.io
+  if [ $? != 0 ]; then
+    err 'Failed to install Docker'
+  fi
+
   if [[ -z "${DOCKER_IMAGE}" ]]; then
     DOCKER_IMAGE="gcr.io/cloud-datalab/datalab:local"
   fi
+  readonly DOCKER_IMAGE
+
   gcloud docker -- pull ${DOCKER_IMAGE}
+  if [ $? != 0 ]; then
+    err 'Failed to pull ${DOCKER_IMAGE}'
+  fi
 
   # For some reason Spark has issues resolving the user's directory inside of
   # Datalab.
   # TODO(pmkc) consider fixing in Dataproc proper.
-  SPARK_CONF='/etc/spark/conf/spark-defaults.conf'
   if ! grep -q '^spark\.sql\.warehouse\.dir=' "${SPARK_CONF}"; then
     echo 'spark.sql.warehouse.dir=/root/spark-warehouse' >> "${SPARK_CONF}"
   fi
 
-  DATALAB_DIR="${HOME}/datalab"
-  mkdir -p "${DATALAB_DIR}"
-
   # Expose every possible spark configuration to the container.
   VOLUMES=$(echo /etc/{hadoop*,hive*,*spark*} /usr/lib/hadoop/lib/{gcs,bigquery}*)
```

Comment:
Can these four lines be pulled out into a `readonly VOLUME` and `readonly VOLUME_FLAGS` at the top? I don't think the `echo` line is necessary because you are running with `set -x`